lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1250,15 +1252,24 @@ def _on_logging_relation_changed(self, event: HookEvent):
                 self._process_logging_relation_changed(relation)
 
     def _on_logging_relation_departed(self, event: RelationDepartedEvent):
-        """Removes alert rules files when consumer charms left the relation with Loki.
+        """Handle departures in related consumers.
 
         Args:
             event: a `CharmEvent` in response to which the Loki
                 charm must update its relation data.
         """
         self._process_logging_relation_changed(event.relation)
 
-    def update_endpoint(self, relation: Relation = None) -> None:
+    def _on_logging_relation_broken(self, event: RelationBrokenEvent):
+        """Handle broken relation in related consumers.
+
+        Args:
+            event: a `CharmEvent` in response to which the Loki
+                charm must update its relation data.
+        """
+        self._process_logging_relation_changed(event.relation)
```

Comment:
@PietroPasotti see evidence for my claim that there is no reason to be so fine-grained in terms of _why_ ingress-provided URLs change.